perf_generic.go

```diff
@@ -0,0 +1,17 @@
+//+build !amd64
+
+package perf
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+// doEnableRunDisable enables the counters, executes f, and disables them. Where
+// possible it is implemented in assembly to minimize non-deterministic
+// overhead. It is assumed that perfFD is known to be a valid file descriptor at
+// the time of the call, no error checking occurs.
+func doEnableRunDisable(fd uintptr, f func()) {
+	unix.Syscall(unix.SYS_IOCTL, fd, uintptr(unix.PERF_EVENT_IOC_ENABLE), 0)
+	f()
+	unix.Syscall(unix.SYS_IOCTL, fd, uintptr(unix.PERF_EVENT_IOC_DISABLE), 0)
```

Comment:
A friend suggested that we might be able to use `RawSyscall` here, to minimize overhead even further. `unix.RawSyscall` is one more branch compared to `syscall.RawSyscall`, so perhaps this could be

`syscall.RawSyscall(unix.SYS_IOCTL, fd, uintptr(unix.PERF_EVENT_IOC_DISABLE), 0)`.

I'm fine with ignoring errors by virtue of the special circumstances we are in.